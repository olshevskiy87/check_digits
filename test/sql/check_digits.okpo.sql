CREATE EXTENSION IF NOT EXISTS check_digits;

-- should be true
select check_digits_okpo('47296611');
select check_digits_okpo('0154489581');

-- should be false
select check_digits_okpo('0193273379');

-- should fail with error
-- by length
select check_digits_okpo('1000073254140');
select check_digits_okpo('308363');
-- by wrong characters
select check_digits_okpo('01G4489S81');
