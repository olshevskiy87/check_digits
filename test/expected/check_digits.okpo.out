CREATE EXTENSION IF NOT EXISTS check_digits;
-- should be true
select check_digits_okpo('47296611');
 check_digits_okpo 
-------------------
 t
(1 row)

select check_digits_okpo('0154489581');
 check_digits_okpo 
-------------------
 t
(1 row)

-- should be false
select check_digits_okpo('0193273379');
 check_digits_okpo 
-------------------
 f
(1 row)

-- should fail with error
-- by length
select check_digits_okpo('1000073254140');
ERROR:  OKPO must contain 8 or 10 digits
select check_digits_okpo('308363');
ERROR:  OKPO must contain 8 or 10 digits
-- by wrong characters
select check_digits_okpo('01G4489S81');
ERROR:  invalid symbol "G"
